system-addon/test/unit/lib/Store.test.js

```diff
@@ -38,55 +49,129 @@ describe("Store", () => {
+        return feed;
+      }
+      store._feedConstructors = {foo: createFeed};
+
+      store.initFeed("foo");
+      store.uninitFeed("foo");
+
+      assert.calledOnce(feed.uninit);
     });
-    it("should add a reference to the store to each feed", () => {
-      const s = {};
+    it("should remove the feed from .feeds", () => {
+      class Foo {}
+      store._feedConstructors = {foo: () => new Foo()};
 
-      store.init([s]);
+      store.initFeed("foo");
+      store.uninitFeed("foo");
 
-      assert.equal(s.store, store, "s.store");
+      assert.isFalse(store.feeds.has("foo"), "foo is not in .feeds");
+    });
+  });
+  describe("maybeStartFeedAndListenForPrefChanges", () => {
+    beforeEach(() => {
+      sinon.stub(store, "initFeed");
+      sinon.stub(store, "uninitFeed");
+    });
+    it("should set the new pref in Preferences to true, if it was never defined", () => {
+      store.maybeStartFeedAndListenForPrefChanges("foo");
+      assert.isTrue(Preferences.get(`${PREF_PREFIX}foo`));
+    });
+    it("should not override the pref if it was already set", () => {
+      Preferences.set(`${PREF_PREFIX}foo`, false);
+      store.maybeStartFeedAndListenForPrefChanges("foo");
+      assert.isFalse(Preferences.get(`${PREF_PREFIX}foo`));
+    });
+    it("should initialize the feed if the Pref is set to true", () => {
+      Preferences.set(`${PREF_PREFIX}foo`, true);
+      store.maybeStartFeedAndListenForPrefChanges("foo");
+      assert.calledWith(store.initFeed, "foo");
+    });
+    it("should not initialize the feed if the Pref is set to false", () => {
+      Preferences.set(`${PREF_PREFIX}foo`, false);
+      store.maybeStartFeedAndListenForPrefChanges("foo");
+      assert.notCalled(store.initFeed);
+    });
+    it("should observe the pref", () => {
+      store.maybeStartFeedAndListenForPrefChanges("foo");
+      assert.calledWith(Preferences.observe, `${PREF_PREFIX}foo`, store._prefHandlers.get(`${PREF_PREFIX}foo`));
+    });
+    describe("handler", () => {
+      let handler;
+      beforeEach(() => {
+        store.maybeStartFeedAndListenForPrefChanges("foo");
+        handler = store._prefHandlers.get(`${PREF_PREFIX}foo`);
+      });
+      it("should initialize the feed if called with true", () => {
+        handler(true);
+        assert.calledWith(store.initFeed, "foo");
+      });
+      it("should uninitialize the feed if called with false", () => {
+        handler(false);
+        assert.calledWith(store.uninitFeed, "foo");
+      });
+    });
+  });
+  describe("#init", () => {
+    it("should set ._feedConstructors to the object passed to init", () => {
+      class Foo {}
+      const feeds = {foo: () => new Foo(), bar: () => new Foo()};
+      store.init(feeds);
+      assert.equal(store._feedConstructors, feeds);
+    });
+    it("should call .maybeStartFeedAndListenForPrefChanges with each key", () => {
+      sinon.stub(store, "maybeStartFeedAndListenForPrefChanges");
```

Comment:
True.  I guess I tend to bias towards always using sandbox except when it's super obvious (eg an anonymous stub is assigned directly to a property), for two reasons: if I always do the conservative thing, it's impossible for me to make that particular thinko, and it tends to be more a bit more robust against code changes or copy-pasting of code.